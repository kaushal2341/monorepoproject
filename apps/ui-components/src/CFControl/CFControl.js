import React from 'react';
import Form from "react-bootstrap/Form";
import PropTypes from "prop-types";

const CFControl = props => {
    const {
        _ref,
        as,
        autoComplete,
        bsPrefix,
        children,
        className,
        defaultValue,
        disabled,
        id,
        isInvalid,
        isValid,
        max,
        min,
        multiple,
        name,
        onChange,
        pattern,
        placeholder,
        plaintext,
        readOnly,
        required,
        rows,
        size,
        type,
        value,
    } = props;
    return (
        <Form.Control
            _ref={_ref}
            as={as}
            autoComplete={autoComplete}
            bsPrefix={bsPrefix}
            children={children}
            className={className}
            defaultValue={defaultValue}
            disabled={disabled}
            id={"fControl_".concat(id)}
            isInvalid={isInvalid}
            isValid={isValid}
            max={max}
            min={min}
            multiple={multiple}
            name={name}
            onChange={onChange}
            pattern={pattern}
            placeholder={placeholder}
            plaintext={plaintext}
            readOnly={readOnly}
            required={required}
            rows={rows}
            size={size}
            type={type}
            value={value}
        />
    )
};

CFControl.propTypes = {
    _ref: PropTypes.oneOfType([
        PropTypes.func, // for legacy refs
        PropTypes.shape({current: PropTypes.instanceOf(Element)})
    ]),
    as: PropTypes.elementType,
    autoComplete: PropTypes.string,
    bsPrefix: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    defaultValue: PropTypes.string,
    disabled: PropTypes.bool,
    id: PropTypes.string,
    isInvalid: PropTypes.bool,
    isValid: PropTypes.bool,
    max: PropTypes.number,
    min: PropTypes.number,
    multiple: PropTypes.bool,
    name: PropTypes.string,
    onChange: PropTypes.func,
    pattern: PropTypes.string,
    placeholder: PropTypes.string,
    plaintext: PropTypes.bool,
    readOnly: PropTypes.bool,
    required: PropTypes.bool,
    rows: PropTypes.string,
    size: PropTypes.string, // 'sm','lg
    type: PropTypes.string,
    value: PropTypes.string,
};

export default CFControl;