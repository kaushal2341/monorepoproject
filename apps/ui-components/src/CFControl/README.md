# CFControl

This is a `Form Control` component made using react-bootstrap's `Form.Control`.

To use it ,first import the component,

```jsx harmony
import {CFControl} from '@cogent/ui-components';
```
Then use it as,

```jsx harmony
<CFControl
   id="text"
   type="email"
   placeholder="Enter your email"
   size="lg"
   required={true}
/>
```
*Note:* While using **CFControl** with **CFormGroup**, either pass only `id` for **CFControl** or only pass `controlId` for **CFormGroup**.
If we pass `controlId` for **CFormGroup**, don't pass `id` for **CFControl** and vice versa.
```
Warning: `controlId` is ignored on `<FormControl>` when `id` is specified.
```
To do:
```jsx harmony
<CFormGroup
    id="test"
    controlId="formHorizontalEmail">
    <CFControl
       type="email"
       placeholder="Enter your email"
       size="lg"
       required={true}
    />
  </CFormGroup>
```

Not to do:

```jsx harmony
<CFormGroup
    id="test"
    controlId="formHorizontalEmail">
    <CFControl
       id="email"
       type="email"
       placeholder="Enter your email"
       size="lg"
       required={true}
    />
  </CFormGroup>
```

## CFControl Props:
```text
 _ref: ReactRef - The FormControl ref will be forwarded to the underlying input element, which means unless as is a composite component, it will be a DOM node, when resolved.
 
as: 'input' | 'textarea' | elementType - The underlying HTML element to use when rendering the FormControl.

autoComplete: string - To prevent auto-complete of CFControl, pass in 'off' to disable autocomplete.Specially to use when using CFControl as password.

bsPrefix: string - defaultValue: {'form-control'} - Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatch for working with heavily customized bootstrap css.

children: node - Children element of the component.

className: string - Css classname for additional styling.

defaultValue: string - Default value for Form Control.

id: string - Uses controlId from <FormGroup> if not explicitly specified.

isInvalid: boolean - Add "invalid" validation styles to the control and accompanying label

isValid: boolean - Add "valid" validation styles to the control.

max: number - Max number allowed when using CFormControl as number input.

min:number - Min number allowed when using CFormControl as number input.

multiple: boolean - When using CFControl as 'select', enables selection of multiple option.

name: string - Name of CFControl(Form Control).

onChange: function - A callback fired when the value prop changes.

placeholder: string - Placeholder for CFControl.

plaintext: boolean - Render the input as plain text. Generally used along side readOnly.

readOnly: boolean - Make the control readonly.

required: boolean - To mark certain field's value compulsory.

rows: string - To specify row width when using as text area.

size: string - Eg.'sm','lg - Input size variants.

type: string - The HTML input type, which is only relevant if as is 'input' (the default).

value: string - controlled by: onChange, initial prop: defaultValue
                The value attribute of underlying input.
```