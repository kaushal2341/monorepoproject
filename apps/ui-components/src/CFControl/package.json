{
  "name": "@cogent/c-f-control",
  "version": "0.0.0",
  "main": "CFControl.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
