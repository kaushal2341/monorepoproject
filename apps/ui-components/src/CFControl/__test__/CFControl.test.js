import CFControl from "../CFControl";

expect.addSnapshotSerializer(enzymeSerializer);

describe('CFControl Component Tests', () => {

    let wrapper, mockFunction;
    beforeEach(() => {
        mockFunction = jest.fn();
        wrapper = shallow(<CFControl id="test" onChange={mockFunction}/>);
    });

    it('should be defined', () => {
        expect(wrapper.find('CFControl')).toBeDefined();
    });

    it('should have only one Form.Control component', () => {
        expect(wrapper.find('#fControl_test').length).toBe(1);
    });

    it('should have all props available', () => {
        let propsAvailable = [
            '_ref',
            'as',
            'autoComplete',
            'bsPrefix',
            'children',
            'className',
            'defaultValue',
            'disabled',
            'id',
            'isInvalid',
            'isValid',
            'max',
            'min',
            'multiple',
            'name',
            'onChange',
            'pattern',
            'placeholder',
            'plaintext',
            'readOnly',
            'required',
            'rows',
            'size',
            'type',
            'value',
        ];
        let propsOfFormControl = Object.keys(wrapper.find('#fControl_test').props());
        propsAvailable.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfFormControl[i])
        ));
    });

    it('should call event handler function onChange', () => {
        wrapper.simulate('change');
        expect(mockFunction).toHaveBeenCalled();
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });

});