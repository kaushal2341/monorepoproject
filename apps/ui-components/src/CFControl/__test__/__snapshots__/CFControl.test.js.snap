// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFControl Component Tests Snapshot Testing should match snapshot 1`] = `
<FormControl
  id="fControl_test"
  onChange={[MockFunction]}
/>
`;
